From 100fbb09ba35a576d8773491b861bf5ee2615ad4 Mon Sep 17 00:00:00 2001
From: Tim Gore <tim.gore@intel.com>
Date: Thu, 17 Dec 2015 15:32:08 +0000
Subject: [PATCH] SQUASHME! [VPG]: drm/i915: fix bug in request_find_by_seqno

The function i195_gem_request_find_by_seqno in i915_gem.c has
a bug caused by it calling i915_seqno_passed with the arguments
the wrong way round, causing the search to end prematurely.

SQUASHME! - This patch should be squashed into the following patch:
	Author: John Harrison <John.C.Harrison@Intel.com>
	Date:   Fri Jan 9 13:32:36 2015 +0000
	Change-Id: Ia80f332a25c2a3dfcaaadcf4c8bfcf718f24cf64
	REVERTME [VPG]: drm/i915: Make native sync work with scheduler

Change-Id: I05594b185b96f4d2059e54dc6c9047c62c5f4fb8
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-12931
Signed-off-by: Tim Gore <tim.gore@intel.com>
Reviewed-on: https://android.intel.com:443/451493
---
 drivers/gpu/drm/i915/i915_gem.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_gem.c b/drivers/gpu/drm/i915/i915_gem.c
index e335cf4..253a559 100644
--- a/drivers/gpu/drm/i915/i915_gem.c
+++ b/drivers/gpu/drm/i915/i915_gem.c
@@ -3111,15 +3111,17 @@ struct drm_i915_gem_request *i915_gem_request_find_by_seqno(struct intel_engine_
 		if (req->seqno == seqno)
 			break;
 
-		if (i915_seqno_passed(seqno, req->seqno)) {
+		if (i915_seqno_passed(req->seqno, seqno)) {
 			DRM_DEBUG_DRIVER("Searching for missing seqno!\n");
-			break;
+			req = NULL;
+			goto unlock;
 		}
 	}
 
 	if (&req->list == &ring->request_list)
 		req = NULL;
 
+unlock:
 	spin_unlock_irqrestore(&ring->reqlist_lock, flags);
 
 	return req;
-- 
1.9.1

